### PR TITLE
LPS-120859 [WIP] As a publication user, I would like to add create a site in a publication

### PR DESCRIPTION
**Pull request by stsquared99**

### CODE REVIEW

File: modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/test/GroupServiceUserSitesGroupsCTTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+			List<Group> groups = _groupService.getUserSitesGroups(
+				TestPropsValues.getUserId(), null, QueryUtil.ALL_POS);
+
+			Assert.assertTrue(
+				groups + " contains " + _group, groups.contains(_group));
+			Assert.assertTrue(
+				groups + " contains " + group, groups.contains(group));
+		}
+
+		List<Group> groups = _groupService.getUserSitesGroups(
```
Comment (stsquared99) [inline modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/test/GroupServiceUserSitesGroupsCTTest.java:88]: Throws NoSuchGroupException

**Comment by shuyangzhou** [comment]
@stsquared99 the test fails with 
```com.liferay.change.tracking.internal.test.GroupServiceUserSitesGroupsCTTest > testGetUserSitesGroups FAILED
    com.liferay.portal.kernel.exception.ModelListenerException: com.liferay.portal.kernel.change.tracking.CTTransactionException: CT transaction validation failure. Nested operation using com.liferay.portal.security.audit.storage.service.impl.AuditEventLocalServiceImpl@56c1317a can only be performed in production mode.
    	at com.liferay.portal.security.audit.event.generators.internal.model.listener.UserGroupRoleModelListener.auditOnCreateOrRemove(UserGroupRoleModelListener.java:82)
    	at com.liferay.portal.security.audit.event.generators.internal.model.listener.UserGroupRoleModelListener.onBeforeCreate(UserGroupRoleModelListener.java:44)
    	at com.liferay.portal.security.audit.event.generators.internal.model.listener.UserGroupRoleModelListener.onBeforeCreate(UserGroupRoleModelListener.java:37)
    	at com.liferay.portal.kernel.service.persistence.impl.BasePersistenceImpl.update(BasePersistenceImpl.java:699)
    	at com.liferay.portal.service.impl.UserGroupRoleLocalServiceImpl.addUserGroupRole(UserGroupRoleLocalServiceImpl.java:51)
    	at com.liferay.portal.service.impl.UserGroupRoleLocalServiceImpl.addUserGroupRoles(UserGroupRoleLocalServiceImpl.java:77)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:498)
    	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:50)
    	at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:69)
    	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
    	at com.liferay.portal.spring.aop.AopInvocationHandler.invoke(AopInvocationHandler.java:49)
    	at com.sun.proxy.$Proxy65.addUserGroupRoles(Unknown Source)
    	at com.liferay.portal.service.impl.GroupLocalServiceImpl.addGroup(GroupLocalServiceImpl.java:458)
    	at com.liferay.portal.service.impl.GroupLocalServiceImpl.addGroup(GroupLocalServiceImpl.java:490)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:498)
    	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:50)
    	at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:69)
    	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
    	at com.liferay.portal.service.ServiceContextAdvice.invoke(ServiceContextAdvice.java:60)
    	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
    	at com.liferay.portal.spring.aop.AopInvocationHandler.invoke(AopInvocationHandler.java:49)
    	at com.sun.proxy.$Proxy38.addGroup(Unknown Source)
    	at com.liferay.portal.kernel.service.GroupLocalServiceUtil.addGroup(GroupLocalServiceUtil.java:79)
    	at com.liferay.portal.kernel.test.util.GroupTestUtil.addGroup(GroupTestUtil.java:113)
    	at com.liferay.portal.kernel.test.util.GroupTestUtil.addGroup(GroupTestUtil.java:54)
    	at com.liferay.portal.kernel.test.util.GroupTestUtil.addGroup(GroupTestUtil.java:50)
    	at com.liferay.change.tracking.internal.test.GroupServiceUserSitesGroupsCTTest.testGetUserSitesGroups(GroupServiceUserSitesGroupsCTTest.java:77)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:498)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:353)
    	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
    	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$1.evaluate(TestExecutorRunnable.java:237)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
    	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
    	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable._execute(TestExecutorRunnable.java:316)
    	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable.run(TestExecutorRunnable.java:95)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: com.liferay.portal.kernel.change.tracking.CTTransactionException: CT transaction validation failure. Nested operation using com.liferay.portal.security.audit.storage.service.impl.AuditEventLocalServiceImpl@56c1317a can only be performed in production mode.
    	at com.liferay.change.tracking.internal.aop.CTTransactionAdvice.invoke(CTTransactionAdvice.java:102)
    	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
    	at com.liferay.portal.spring.aop.AopInvocationHandler.invoke(AopInvocationHandler.java:49)
    	at com.sun.proxy.$Proxy670.addAuditEvent(Unknown Source)
    	at com.liferay.portal.security.audit.storage.internal.AuditEventManagerImpl.addAuditEvent(AuditEventManagerImpl.java:40)
    	at com.liferay.portal.security.audit.router.internal.PersistentAuditMessageProcessor.doProcess(PersistentAuditMessageProcessor.java:79)
    	at com.liferay.portal.security.audit.router.internal.PersistentAuditMessageProcessor.process(PersistentAuditMessageProcessor.java:47)
    	at com.liferay.portal.security.audit.router.internal.DefaultAuditRouter.route(DefaultAuditRouter.java:94)
    	at sun.reflect.GeneratedMethodAccessor176.invoke(Unknown Source)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:498)
    	at com.liferay.portal.kernel.messaging.proxy.ProxyRequest.execute(ProxyRequest.java:82)
    	at com.liferay.portal.kernel.messaging.proxy.ProxyMessageListener.receive(ProxyMessageListener.java:58)
    	at com.liferay.portal.kernel.messaging.InvokerMessageListener.receive(InvokerMessageListener.java:74)
    	at com.liferay.portal.messaging.internal.sender.DirectSynchronousMessageSender.send(DirectSynchronousMessageSender.java:67)
    	at com.liferay.portal.kernel.messaging.proxy.BaseProxyBean.synchronousSend(BaseProxyBean.java:60)
    	at com.liferay.portal.kernel.messaging.proxy.MessagingProxyInvocationHandler.invoke(MessagingProxyInvocationHandler.java:55)
    	at com.sun.proxy.$Proxy252.route(Unknown Source)
    	at com.liferay.portal.security.audit.event.generators.internal.model.listener.UserGroupRoleModelListener.auditOnCreateOrRemove(UserGroupRoleModelListener.java:79)
    	... 59 more

1 test completed, 1 failed
```
This is not what you meant to show, right? This failure is a missing CT enabled service.

**Comment by stsquared99** [comment]
Hi @shuyangzhou, that's weird. Is this is a clean bundle (ct not enabled manually) with the layout seo code deployed? I just ran the test off of a clean ant all and the layout seo code deployed and I didn't get that exception, only the nosuchgroupexception.

**Comment by stsquared99** [comment]
Hey @shuyangzhou, I was able to meet with Preston and he gave me a direction for how to handle this. Thanks!

**Pull request closed without merging**